Docs: Move DevEnv to own section for docs

mkaz opened the pull request:
## Description

This moves the Development Environment documentation out of the Create a Block tutorial and to its own section. This allows for easier linking from elsewhere. Plus allows for simplifying the Create a Block tutorial, with linking over the the DevEnv as a pre-requisitite.

It also gives a section to expand more on the Development Environment setup. I created it as its own directory so if we want to break each piece out—Node, WordPress Environment, and Code Editor—to its own page, there is space for it to grow.

This new development environment is related to the [Contributors Getting Started Guide](https://developer.wordpress.org/block-editor/contributors/develop/getting-started/). The contributor guide docs are specific to contributing to Gutenberg, whereas the guide in this PR is specific to plugin developers extending the block editor.

See: #22831 

## How has this been tested?

Documentation, confirm links work and docs make sense.

View the [rendered version on branch here](https://github.com/WordPress/gutenberg/tree/docs/cab-devenv/docs/designers-developers/developers/tutorials/devenv)

## Types of changes

Documentation

mkaz committed:
Move devenv to own section

mkaz committed:
Update links to new devenv section

mkaz committed:
Add devenv to docs TOC

paaljoachim commented on docs/designers-developers/developers/tutorials/create-block/readme.md:
```diff
@@ -20,13 +20,18 @@ You now need to activate the plugin from inside wp-admin plugins page.
 
+## Prerequisities
+
+The first thing you need is a development enviornment and tools. This includes setting up your WordPress environment, Node, NPM, and your code editor. If you need help, see the [setting up your development environment documentation](/docs/designers-developers/developers/tutorials/devenv/readme.md).
```
The first thing you need is a development environment and tools. This includes setting up your WordPress environment, Node, NPM, and your code editor. If you need help, see the [setting up your development environment documentation](/docs/designers-developers/developers/tutorials/devenv/readme.md).

paaljoachim commented:
Title: Docs: Move DevEnv to own section for docs
-->
Title: Docs: Move DevEnv to its own section in the docs.

---
"This moves the Development Environment documentation out of the Create a Block tutorial and to its own section. This allows for easier linking from elsewhere. Plus allows for simplifying the Create a Block tutorial, with linking over the the DevEnv as a pre-requisitite."
--->
This moves the Development Environment documentation out of the Create a Block tutorial. It brings greater freedom in linking to other documentation. DevEnv can also become a pre-requisitite for a simplified Create a Block tutorial.
---------
"It also gives a section to expand more on the Development Environment setup. I created it as its own directory so if we want to break each piece out—Node, WordPress Environment, and Code Editor—to its own page, there is space for it to grow."
--->
We can also expand the Development Environment setup to include additional sections. (No need to include the second sentence. As we have already said that it can be expanded.)
-------
"This new development environment is related to the Contributors Getting Started Guide. The contributor guide docs are specific to contributing to Gutenberg, whereas the guide in this PR is specific to plugin developers extending the block editor."
--->
There are two development environment guides. 
The one we are working on here is for plugin developers to extend the block editor.
The other is related to Getting Started Guide for Contributors.

paaljoachim commented:
Going through Development Environment docs.

"A development enviornment (typo) is a catch-all term for the setup on your computer to work." 
-->
A development environment in this context is a catch-all term for developing with WordPress and Gutenberg.

"The script npx is installed with npm and is used to run packages not yet installed, we will use this to bootstrap a block."
-->
This is a pretty complex sentence and could be broken down + add some more details to it to make it easier to understand for a non developer.


"Here are the quick instructions to install using nvm, ..."
-->
Here are the quick instructions for the Mac to install using nvm.



-----

Run the following on the command-line to install nvm:
`curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`

A message might show up:

<img width="1175" alt="Screen Shot 2020-07-05 at 15 00 59" src="https://user-images.githubusercontent.com/5323259/86533279-5b275200-bed0-11ea-9faa-b35acbf129e8.png">

In terminal it might say this:
```
=> xcode-select: note: no developer tools were found at '/Applications/Xcode.app', requesting install. 
Choose an option in the dialog to download the command line developer tools.
Failed to clone nvm repo. Please report this!
```

Click Install to open the Mac Automatic Update, and the command line developer tools will be downloaded and installed.


Run the Node Version Manager install command line again:
`curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`

This time there should be no error message as the Mac command line developer tools have been installed.


This is what I saw:

```
paaljoachimromdahl@Paals-MacBook-Pro ~ % curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 13527  100 13527    0     0  67635      0 --:--:-- --:--:-- --:--:-- 67635
=> Downloading nvm from git to '/Users/paaljoachimromdahl/.nvm'
=> Cloning into '/Users/paaljoachimromdahl/.nvm'...
remote: Enumerating objects: 290, done.
remote: Counting objects: 100% (290/290), done.
remote: Compressing objects: 100% (257/257), done.
remote: Total 290 (delta 35), reused 97 (delta 20), pack-reused 0
Receiving objects: 100% (290/290), 163.27 KiB | 320.00 KiB/s, done.
Resolving deltas: 100% (35/35), done.
=> Compressing and cleaning up git repository

=> Appending nvm source string to /Users/paaljoachimromdahl/.zshrc
=> Appending bash_completion source string to /Users/paaljoachimromdahl/.zshrc
=> Close and reopen your terminal to start using nvm or run the following to use it now:

export NVM_DIR="$HOME/.nvm"
[ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
[ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
paaljoachimromdahl@Paals-MacBook-Pro ~ % 
```

After installing nvm, you now need to install the latest version of node, run:
`nvm install --lts`

This is the message I received:
```
paaljoachimromdahl@Paals-MacBook-Pro ~ % nvm install --lts
zsh: command not found: nvm
paaljoachimromdahl@Paals-MacBook-Pro ~ %
```

If there is an error running the above command line code on the Mac then you might need to create the required file zshrc. To do so type `touch ~/.zshrc`
It is fine to run if the file already exists. Note, `~/` is a shortcut to your home directory.

This is what I noticed in terminal:
```
paaljoachimromdahl@Paals-MacBook-Pro ~ % nvm install --lts
zsh: command not found: nvm
paaljoachimromdahl@Paals-MacBook-Pro ~ % touch ~/.zshrc
paaljoachimromdahl@Paals-MacBook-Pro ~ % nvm install --lts
zsh: command not found: nvm
paaljoachimromdahl@Paals-MacBook-Pro ~ % node -v
zsh: command not found: node
paaljoachimromdahl@Paals-MacBook-Pro ~ % npm -v
zsh: command not found: npm
paaljoachimromdahl@Paals-MacBook-Pro ~ % 
```

So here it stopped up, and I am not certain where to go from here.

I am running a fresh reformated Macbook Pro running OSX Catalina.

gziolo commented on docs/designers-developers/developers/tutorials/create-block/wp-plugin.md:
```diff
@@ -48,7 +48,7 @@ The script created a PHP file `gutenpride.php` that is the main plugin file. At
 -or-
 
-(3B) If you are using `wp-env`, see [Development Environment setup](devenv.md), then you should now run from inside the `gutenpride` directory:
+(3B) If you are using `wp-env`, see [Development Environment setup](/docs/designers-developers/developers/tutorials/devenv/devenv.md), then you should now run from inside the `gutenpride` directory:
```
```suggestion
(3B) If you are using `wp-env`, see [Development Environment setup](/docs/designers-developers/developers/tutorials/devenv/readme.md), then you should now run from inside the `gutenpride` directory:
```

gziolo commented on docs/designers-developers/developers/tutorials/readme.md:
```diff
@@ -1,15 +1,17 @@
 # Tutorials
 
-* See the [Getting Started with JavaScript Tutorial](/docs/designers-developers/developers/tutorials/javascript/readme.md) to learn about how to use JavaScript within WordPress.
+-   First things first, see [setting up your development environment](/docs/designers-developers/developers/tutorials/javascript/readme.md) for the tools and setup you need to extend the block editor.
```
```suggestion
-   First things first, see [setting up your development environment](/docs/designers-developers/developers/tutorials/devenv/readme.md) for the tools and setup you need to extend the block editor.
```

gziolo commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
```diff
@@ -1,6 +1,8 @@
 # Development Environment
```
I see that this file is named `readme.md`. I double-checked and it works on GitHub the same way as `README.md` – it displays its content by default when entering the folder:
https://github.com/WordPress/gutenberg/tree/1e0bda1e09d05eab04149171847cc0ee05e99520/docs/designers-developers/developers/tutorials/devenv

So I'm curious why we use capitalized version in all places next to the code :)

mkaz replied:
That's a good question, I don't know why or how it started.

mkaz committed:
Apply suggestions from code review

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

mkaz commented:
@paaljoachim Thanks for the feedback, I need to go through each still but for future reference, Github has a suggestion and review tool that allows submitting feedback in context with the lines of code. 

See [this Github documentation for more info](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/reviewing-proposed-changes-in-a-pull-request)

Using the suggestion tool makes line edits, spelling and also associates your name with the change.

mkaz committed:
Update devenv with review feedback

mkaz commented:
@paaljoachim Thanks for the feedback, I updated the docs addressing your points. I modified slightly so it includes instructions for additional environments besides the Mac.

Also, to note the alternative environment is only for Docker, the headings have it under that section. Node is still required and there is no alternative, only different methods to installing which are already linked to above.

mkaz committed:
Switch to image tag

mkaz committed:
Add link in JS tutorial to new dev env section

Adds an additional link in JS tutorial to the new development
environment section, with regards to installing using nvm.

Plus includes a bit of markdown linting.

paaljoachim commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
```diff
@@ -18,20 +20,41 @@ For Windows, or alternative installs, you can [download a Nodejs installer](http
 
 Here are the quick instructions to install using nvm, see the [full installation instructions](https://github.com/nvm-sh/nvm#installing-and-updating) for additional details.
 
```
Here are the quick instructions for Windows and Mac to install using nvm, see the [full installation instructions](https://github.com/nvm-sh/nvm#installing-and-updating) for additional details.
(As we need to let the user know which OS we are giving quick instructions to.)

mkaz replied:
The instructions also work for Linux, so it is for all.

paaljoachim commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
```diff
@@ -18,20 +20,41 @@ For Windows, or alternative installs, you can [download a Nodejs installer](http
+
+First, try quitting and restarting your terminal to pick up the installed config.
+
+If the restating did not resolve, you might need to create the default profile file.
```
restating = restarting

mkaz replied:
Fixed 👍

noahtallen commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
````diff
@@ -52,20 +75,28 @@ The WordPress [wp-env package](https://www.npmjs.com/package/@wordpress/env) let
+The `wp-env` script is used to create a Docker WordPress environment, to start use:
+
+```sh
+wp-env start
````
I'm not super familiar with this file, but it may be helpful to mention:
- configuring wp-env with the `.wp-env.json` file for project-specific settings
- If you're developing one plugin/theme primarily, you should run `wp-env start` from plugin/theme directory. (if you do so, it will mount the plugin/theme automatically)

If you want one environment for all your plugins/themes, you'd just have to create a `.wp-env.json` in some arbitrary location and add the plugins/themes to it. (then, you have to run `wp-env start` from the same directory as that config file

mkaz replied:
Thanks, I've updated with this info. 👍 

This page is a generic page for getting a development environment setup, the intended audience is a plugin or theme developer. So this is right on point, I appreciate the additional info for using multiple plugins/themes.

paaljoachim replied:
```
wp-env --version
> 1.6.0
```

The wp-env script is used to create a Docker WordPress environment, to use start from your plugin directory—if you are following the create block tutorial, this would be in the generated directory.
`wp-env start`

It is confusing going from working in terminal to suddenly mentioning the plugin directory. 
Where is the plugin directory? We need a step that goes from checking the wp-env to locating the plugin directory and then running `wp-env start`. This needs clarification.


"...if you are following the create block tutorial, this would be in the generated directory."
This seems out of context. As the context is setting up a local dev environment.

paaljoachim commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
````diff
@@ -52,20 +75,28 @@ The WordPress [wp-env package](https://www.npmjs.com/package/@wordpress/env) let
+```sh
+wp-env start
+```
+
````
This is what I see in terminal:
paaljoachimromdahl@Paals-MacBook-Pro ~ % wp-env --version
1.6.0
```
paaljoachimromdahl@Paals-MacBook-Pro ~ % wp-env start
✖ No .wp-env.json file found at '/Users/paaljoachimromdahl/.wp-env.json' and could not determine if '/Users/paaljoachimromdahl' is a WordPress installation, a plugin, or a theme.
paaljoachimromdahl@Paals-MacBook-Pro ~ % 
```

I noticed I have installed Docker but did not start it up. I started up Docker. It asked for privileged access.

<img width="622" alt="Screen Shot 2020-07-08 at 20 52 46" src="https://user-images.githubusercontent.com/5323259/86958823-0a785900-c15d-11ea-9e9a-3f16acb51eb4.png">

Clicking Ok.

<img width="464" alt="Screen Shot 2020-07-08 at 20 53 58" src="https://user-images.githubusercontent.com/5323259/86958916-372c7080-c15d-11ea-9df3-8ee5e078ac23.png">

Docker now starts up. Saying:
"Get started with Docker in a few easy steps."

There is a tutorial there which I clicked next to get into. I am not sure how it is related to wp-env.

<img width="1061" alt="Screen Shot 2020-07-08 at 20 56 11" src="https://user-images.githubusercontent.com/5323259/86959103-85417400-c15d-11ea-9bff-b5d5bb3c3381.png">

I went through the tutorial and end up with this screen.
<img width="1055" alt="Docker-after-tut-screen" src="https://user-images.githubusercontent.com/5323259/86959276-d487a480-c15d-11ea-88cc-37e8624b45b5.png">

noahtallen replied:
The fix for that is related to my earlier comment; basically wp-env needs to be run with either a local config file or from the directory of plugin/theme/core

mkaz replied:
See the error message, it is trying to start in a plugin directory (or find .wp-env.json)
So you need to run it from inside the directory generated using:

`npx @wordpress/create-block gutenpride`

You would cd into `gutenpride` directory and then run `wp-env start`

The additional comments I just added from @noahtallen might help clarify.

paaljoachim commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
````diff
@@ -52,20 +75,28 @@ The WordPress [wp-env package](https://www.npmjs.com/package/@wordpress/env) let
+```
+
+You can access your environment in your browser at: [http://localhost:8888/], the default username is `admin` and default password is `password`. For more information controlling the Docker enivironment see the [@wordpress/env package readme](https://github.com/WordPress/gutenberg/blob/master/packages/env/README.md).
+
````
I believe we have a missing step as I have gone through the various steps and the last step I did was 
`wp-env start` mentioned in my comment above.

mkaz replied:
I'll update to make clear it should be a plugin, when moved out of the tutorial that part got lost. 👍

mkaz replied:
Updated. Let me know if that clarifies

paaljoachim replied:
It did not clarify, as the next step is still missing. There are two options that confuse. There should not be an option. There should just be a next step is add .wp-env.json to terminal to....

As the user is here to setup a local WordPress dev site. Developing a plugin or contributing to WordPress/Gutenberg does not matter at this stage as the only thing we are doing is creating a foundation to develop from. At the end of this setup your WordPress development site tutorial then the user can be guided to Create a Block, Contribute to Gutenberg, Contribute to WordPress etc tutorials. There is no need as I see it to mention it earlier. Creating a foundation should be the same for all of them. One needs to be guided step by step with the most used/broad option to setup the foundation.

mkaz committed:
Update with review comments, typos

mkaz committed:
Clarify wp-env start is from plugin directory

paaljoachim commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
```diff
@@ -1,14 +1,16 @@
 1. Node/NPM Development Tools
 2. WordPress Development Site
 3. Code Editor
 
```
**The short guide.**

In terminal - Mac, Windows or Linux.
Install nvm:
`curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash`
Quit and restart terminal.
Install node:
`nvm install --lts`

Download and install Docker. Startup Docker (?) ...

In terminal.
`npm -g install @wordpress/env`
Check to see it has been installed:
`wp-env --version`

Do something (?)
Then in terminal:
`wp-env start`

You should now have a local development environment in place. If something does not work then check the uninstall guide to remove and start over. - @mkaz and @noahtallen  it would be good to have a section for uninstalling all that we have installed. To start over if needed.

mkaz replied:
The do something step is from the Create a Block tutorial, on generating the plugin directory see [wp-plugin.md](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/tutorials/create-block/wp-plugin.md). 

```
npx @wordpress/create-block gutenpride
cd gutenpride
wp-env start
```

The create block script creates a plugin directory, you run the wp-env start from within that directory.

noahtallen replied:
you can uninstall wp-env like so:
- run `wp-env destroy` to destroy the local environment data for a specific projest
- run `npm uninstall -g @wordpress/env` will uninstall the npm package, but will not remove any environment data

paaljoachim replied:
The do something step is confusing as we are having nice and logical steps on what to do in terminal to setup a local WordPress development. I expect the do something step will then create the local WordPress development site which we can then easily access. I am saying the same thing just below. We should not refer to another tutorial as that information can come later.

paaljoachim replied:
Thanks Noah! Btw how do we test to see that all of it is actually uninstalled?
Having uninstall information and checking to see that it is uninstalled would be helpful.

noahtallen replied:
If it is actually uninstalled, then `wp-env start` will result in an error like "command not found". Additionally, if you destroyed your environment, the command `docker ps -a`, which shows a list of local environments, will not display environments associated with the local environment that was deleted.

Verifying that the environment was properly uninstalled is kind of complicated because you have to understand how Docker works and how wp-env works under the hood, since several different things are cleaned up when you run `wp-env destroy`

paaljoachim replied:
I am thinking as long as one gets the error messages one is supposed to get then that would be the check that it is uninstalled.

paaljoachim commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
````diff
@@ -52,20 +75,32 @@ The WordPress [wp-env package](https://www.npmjs.com/package/@wordpress/env) let
+```
+
+The `wp-env` script is used to create a Docker WordPress environment, to use start from your plugin directory—if you are following the create block tutorial, this would be in the generated directory.
+
````
It is confusing going from working in terminal to suddenly mentioning the plugin directory.
Where is the plugin directory? We need a step that goes from checking the wp-env to locating the plugin directory and then running wp-env start. This needs clarification.

"...if you are following the create block tutorial, this would be in the generated directory."
This seems out of context. As the context is setting up a local dev environment.

We need a bridge going from checking the wp-env version to wp-env script.

(I added a similar comment in the below solved comment thread, but as it got hidden I am adding it again here.)

mkaz replied:
I'm not sure setting up a dev environment with no context makes much sense. There is a reason you are setting up an environment, the most likely is for developing a plugin or theme. If you are following the create a block tutorial it provides the context [in that tutorial here](https://github.com/WordPress/gutenberg/blob/master/docs/designers-developers/developers/tutorials/create-block/wp-plugin.md).

Otherwise, you might of come to this page wanting to setup an environment to create a plugin, so you would have that in mind already.

paaljoachim replied:
We has this introduction:
"The development environment setup guide is for setting up your local development environment for JavaScript development; creating plugins and tools for extending WordPress and the block editor."

This is how I understand it:
Setup a local WordPress development site for creating own plugin and/or contributing to WordPress/Gutenberg. (That is the context.) 

"If you are following the create a block tutorial it provides the context in that tutorial here."
As the doc we are working on right now is the very first tutorial for setting up a local dev site I find it confusing that someone would be following the next tutorial in the series so to speak. As the tutorial we have now would be a prerequisite for the block tutorial. Having a side path to tutorial 2 seems strange at this stage.

gziolo commented on docs/designers-developers/developers/tutorials/create-block/readme.md:
```diff
@@ -20,13 +20,18 @@ You now need to activate the plugin from inside wp-admin plugins page.
-8. [Finishing Touches](finishing.md)
+The create a block tutorials breaks down to the following sections.
+
+1. [WordPress Plugin](wp-plugin.md)
```
Don't we need full URLs here to make sure they work at https://developer.wordpress.org/block-editor/?

```suggestion
1. [WordPress Plugin](/docs/designers-developers/developers/tutorials/create-block/wp-plugin.md)
```

and so on

gziolo commented on docs/designers-developers/developers/tutorials/create-block/readme.md:
```diff
@@ -20,13 +20,18 @@ You now need to activate the plugin from inside wp-admin plugins page.
 
 After activated, go to the block editor and use the inserter to search and add your new block.
 
+## Prerequisities
```
We also would need to update line 5 in this document as it's no longer 100% correct:

> The tutorial includes setting up your development environment, tools, and getting comfortable with the new development model. If you are already comfortable, try the quick start below, otherwise step through whatever part of the tutorial you need.

It might be better to move `Prerequisities` before `Quick Start` as well.

mkaz replied:
Moved it above 👍

gziolo commented on docs/designers-developers/developers/tutorials/devenv/readme.md:
````diff
@@ -52,20 +75,32 @@ The WordPress [wp-env package](https://www.npmjs.com/package/@wordpress/env) let
 ```
 
-### Alternatives
+You can access your environment in your browser at: [http://localhost:8888/], the default username is `admin` and default password is `password`. For more information controlling the Docker enivironment see the [@wordpress/env package readme](https://github.com/WordPress/gutenberg/blob/master/packages/env/README.md).
````
```suggestion
You can access your environment in your browser at: [http://localhost:8888/](http://localhost:8888/), the default username is `admin` and default password is `password`. For more information controlling the Docker environment see the [@wordpress/env package readme](/packages/env/README.md).
```

gziolo commented on docs/designers-developers/developers/tutorials/readme.md:
```diff
@@ -1,15 +1,17 @@
+-   The [Blocks Tutorial](/docs/designers-developers/developers/tutorials/block-tutorial/readme.md) is the best place to start to learn more about block creation.
 
-* Check out the [Notices Tutorial](/docs/designers-developers/developers/tutorials/notices/README.md) to learn how to display informational UI at the top of the editor.
+-   See the [Meta Boxes Tutorial](/docs/designers-developers/developers/tutorials/metabox/readme.md) for new ways of extending the editor storing and using post meta data.
```
Unrelated to this PR, but can we move Meta Boxes Tutorial further? :)

mkaz committed:
Apply suggestions from code review

Co-authored-by: Greg Ziółkowski <grzegorz@gziolo.pl>

mkaz committed:
Move pre-requisites above quick start

mkaz committed:
Update links to use full path

paaljoachim commented:
I think there is some misunderstanding right now. I am kinda saying the same thing over and over again in my direct file comments.

I am still stuck at

```
paaljoachimromdahl@Paals-MacBook-Pro ~ % wp-env start
✖ No .wp-env.json file found at '/Users/paaljoachimromdahl/.wp-env.json' and could not determine if '/Users/paaljoachimromdahl' is a WordPress installation, a plugin, or a theme.
paaljoachimromdahl@Paals-MacBook-Pro ~ % 
```

I have mentioned these things in various ways...
I need the next logical step to move onward in this step by step tutorial.
One minded focus in a step and then one can have some additional information as long as it goes along with the steps in a logical way. If it creates confusion like it does for me at present time, then rethink the placement and the information. 
As in should the information be rephrased and perhaps added later on in the tutorial?
Can the information be cut out? Would it lose anything?

This documentation is about as I see it creating a foundation for a user to develop their own plugin and/or contribute to WordPress/Gutenberg. At the end of the tutorial options for going to Create A Block, Contribute to WordPress Core and/or Gutenberg should be added. As the base foundation for developing with WordPress is in place.

I just had to add the above comment as a comment directly to this issue, as I feel that comments in the file can get a bit lost.

mkaz commented:
This page is a supplemental page to the Create a Bloc tutorial, it was moved out so we can link to it from other tutorials and expand on it the section with additional info that any other tutorial can reference.

The command you are trying to run must be run from a plugins directory. The error message says this also.
It looks like you have the environment setup, it just needs to be started from a plugins directory.

I'll update the instructions but once you have wp-env installed and confirmed you can do:

```
npx @wordpress/create-block gutenpride
cd gutenpride
wp-env start
```

paaljoachim commented:
Version/iteration 2 is good to go. I am looking forward to continuing with testing version/iteration 3.

paaljoachim reviewed (approved):
This is a good version/iteration number 2. Let's continue onward with version/iteration 3.
